openedx/core/djangoapps/user_authn/views/login.py

```diff
@@ -367,9 +367,11 @@ def login_user(request):
 # to get a CSRF token before we need to refresh adds too much
 # complexity.
 @csrf_exempt
-@login_required
 @require_http_methods(['POST'])
 def login_refresh(request):
+    if not request.user.is_authenticated or request.user.is_anonymous:
+        return JsonResponse('Unauthorized', status=401)
```

Comment:
I imagine that this is the crux of the fix, right?  👍

Reply:
Correct, as well as the removal of the `login_required` decorator above. We need to return a 401 instead of redirecting to the login page.